### PR TITLE
Setup schema validation

### PR DESCRIPTION
**Pull request by oschusler**
Additional schemas are added. These schemas are also tested via the "check-jsonschema" python library.

**Note**: The base reply message is changed. Error messages used to be reported via `payload`. This made validation difficult, since the type of payload needed to be `string` in some cases and `object` in other cases. The proposal is to create an `error` property for error messages, and leave `payload` for information for the success reply.

### CODE REVIEW

**Comment by klkl0808** [comment]
Hi Olaf,

I understand the problem with the different data types in the payload. This is indeed not ideal.

But I wonder if we need a separate, optional key in the reply message. Have you considered encoding the error message in the payload dict? I see the following advantages:
- it keeps the number of keys in the message low and avoids optional keys,
- using the payload to transmit details of an error response makes sense semantically,
- it is easier to extend in the future by adding a key to the dict (eg. an error code that is different from the error message).

Would this approach also solve the initial problem?

**Comment by oschusler** [comment]
In principle, I don't have too strong an opinion on this. My only wish is that we don't switch the type of a property. Full disclosure btw. I borrowed the idea from HTTP headers and [JSend](https://github.com/omniti-labs/jsend).

@mjhaye what is your take on this?

**Comment by mjhaye** [comment]
> In principle, I don't have too strong an opinion on this. My only wish is that we don't switch the type of a property. Full disclosure btw. I borrowed the idea from HTTP headers and [JSend](https://github.com/omniti-labs/jsend).
> 
> @mjhaye what is your take on this?

I also do not have a strong opinion. My preference would be to simply send back the error message in the status. So if status != success then it is the error message. Quite similar to the semantics of shell exit codes.

**Comment by klkl0808** [comment]
3 people, 3 opinions :-)

I'm not too fond of the shell-exit-codes suggestion, because it is not explicit and not difficult to extend. There is certainly a benefit in using a "standard" like JSend, but we don't follow JSend anywhere else. So the advantages of my proposal above still sound convincing. Are there any problems that I don't see?

Anyway, I could also live with Olaf's first suggestion if you prefer that one.

**Comment by oschusler** [comment]
I consulted two colleagues. They thought the option put forth by Konstantin would be best. Apparently, this is also how HTTP messages work (and not the message as I proposed it).

I already made the necessary changes. @mjhaye and @klkl0808, if you can review the changes we can merge this PR again

**Review by mjhaye — APPROVED** [review]
The ticket is a lot bigger than the interface change wrt status discussed. For me the latter is ok and the rest I did not review.

File: schemas/publish_dynamic/request.schema.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "title": "Publish dynamic request",
```
Comment (klkl0808) [inline schemas/publish_dynamic/request.schema.json:2]: I find it a bit confusing that it says "request" here. Also in the file name of the schema.

I understand that you re-purpose the base-request schema, but putting the word "request" here, can cause confusion with `trigger_publish`.

**Comment by klkl0808** [comment]
> The ticket is a lot bigger than the interface change wrt status discussed. For me the latter is ok and the rest I did not review.

I was also a bit surprised. I skimmed the schemas and only had one comment. I haven't reviewed them very thoroughly, but they generally look fine to me.

**Review by klkl0808 — APPROVED** [review]

File: schemas/publish_dynamic/request.schema.json (hunk 1 above)
Comment (oschusler) [reply schemas/publish_dynamic/request.schema.json:2]: As discussed; will be done in the next PR

**Review by HarmMunk-TNO — APPROVED** [review]

**Pull request merged**